# `hypothesis`

In [12]:
 from scipy import stats
import numpy as np

# one sample t-test

### prob 1:
`  does the average test score of a class differ significantly from the national average score of 70  `

In [3]:
#sample space of students score
test_score = np.array([78,74,69,80,83,73,68,72,77,82])

In [76]:
#to perform one sample t-test , 
t_stat,p_value= stats.ttest_1samp(test_score,70)

In [78]:
print("t_stat:",t_stat)
print("p_value:",p_val)

t_stat: 3.384456448906594
p_value: 7.951168517443613e-22


In [15]:
#interpretation
if p_value < 0.05:
    print("reject the null hypo : the cls avg is significantly different from 70")
else:
    print("fail to reject the null hypo : the cls avg is not significantly different from 70")

reject the null hypo : the cls avg is significantly different from 70


#### Problem 2: Testing average daily sales of a product

Hypothesis: The average daily sales of a product is 100 units.
Sample Data: Daily sales over 14 days

sales = np.array([98, 102, 100, 97, 103, 101, 99, 100, 101, 98, 104, 96, 102, 100])

In [24]:
# given sample space
d_sale=np.array([98,102,100,97,103,101,99,100,101,98,104,96,102,100])
#Hypothesis: The average daily sales of a product is 100 units.
claimed_mean=100
#to calculate one sample t-test
t_stat,p_val=stats.ttest_1samp(d_sale,100)

In [26]:
print("t_stat:",t_stat)
print("p_value:",p_val)

t_stat: 0.11606680227641673
p_value: 0.9093734022930344


In [28]:
#interpretation
if p_value < 0.05:
    print("reject the null hypo : the avg daily is significantly different from 100")
else:
    print("fail to reject the null hypo : the avg daily is not significantly different from 100")

reject the null hypo : the avg daily is significantly different from 100


#### Problem 3 :  Mean Customer Satisfaction Rating

A company claims that the average customer satisfaction rating is 4.5 out of 5. Test this claim using a sample of customer satisfaction ratings.

Sample data: customer satisfaction ratings (out of 5)

satisfaction_ratings = [4.3, 4.6, 4.5, 4.4, 4.7, 4.8, 4.2, 4.5, 4.6, 4.3]

In [35]:
#sample space
s_rating=np.array([4.3,4.6,4.5,4.4,4.7,4.8,4.2,4.5,4.6,4.3])
#A company claims that the average customer satisfaction rating is 4.5 out of 5
claimed_mean=4.5
#to calculate one sample t-test
t_stat,p_val=stats.ttest_1samp(s_rating,4.5)

In [37]:
print("t_stat:",t_stat)
print("p_value:",p_val)

t_stat: -0.16539535392598792
p_value: 0.8722883630840366


In [50]:
#interpretation
if p_value < 0.05:
    print("reject the null hypo : the avg rating is significantly different from 4.5")
else:
    print("fail to reject the null hypo : the avg rating is not significantly different from 4.5")

reject the null hypo : the avg rating is significantly different from 4.5


#### prb 4:Website Load Time
The IT department states that the average load time of a webpage is 2.5 seconds. Verify this claim using a sample dataset of load times.

Sample data: webpage load times in seconds

load_times = [2.4, 2.6, 2.5, 2.7, 2.3, 2.4, 2.6, 2.5, 2.4, 2.7]

In [46]:
#sample space
l_time=np.array([2.4,2.6,2.5,3.7,2.3,2.4,2.6,2.5,2.4,2.7])
#IT department states that the average load time of a webpage is 2.5 seconds.
claimed_mean=2.5
#to calculate one sample t-test
t_stat,p_val=stats.ttest_1samp(l_time,claimed_mean)

In [48]:
print("t_stat:",t_stat)
print("p_value:",p_val)

t_stat: 0.8669214468630094
p_value: 0.4085029635431019


In [52]:
#interpretation
if p_value < 0.05:
    print("reject the null hypo : the avg timing is significantly different from 2.5")
else:
    print("fail to reject the null hypo : the avg timing is not significantly different from 2.5")

reject the null hypo : the avg timing is significantly different from 2.5


#### prb 5

Problem 5: Testing average weight of athletes


 Hypothesis: The average weight of athletes is 75 kg.
 Sample Data: Weights of 15 athletes (in kg)


weights = np.array([74, 76, 75, 78, 73, 77, 74, 76, 72, 75, 78, 74, 73, 77, 76])

In [14]:
#sample space
a_wt=np.array([74,76,75,78,73,77,74,76,72,75,78,74,73,77,76])
#Hypothesis: The average weight of athletes is 75 kg
claimed_mean = 75
#to calculate one sample t test
t_stat,p_val=stats.ttest_1samp(a_wt,claimed_mean)
print("t_stat:",t_stat)
print("p_value:",p_val)

t_stat: 0.4165977904505369
p_value: 0.6832893490976499


In [57]:
#interpretation
if p_value < 0.05:
    print("reject the null hypo : the avg wt is significantly different from 75")
else:
    print("fail to reject the null hypo : the avg wt is not significantly different from 75")

reject the null hypo : the avg wt is significantly different from 75


 # independent two sample t-test

`do the math scores of two independent groups of students differ significantly `

In [86]:
# given sample space of two groups
grp1_score=np.array([82,76,79,85,88,90,83])
grp2_score=np.array([72,74,70,65,68,75,77])
#to calculate two independent sample t-test
t_stat,p_val=stats.ttest_ind(grp1_score,grp2_score)
print("t_statistic :",round(t_stat,4))
print("p_value :",round(p_val,5))

t_statistic : 4.8097
p_value : 0.00043


In [90]:
#interpretation
if p_value < 0.05:
    print("reject the null hypo : the mean of the two groups is  significantly different from each other")
else:
    print("fail to reject the null hypo : the mean of the two groups is not significantly different from each other")

reject the null hypo : the mean of the two groups is  significantly different from each other


#### prob 2:Comparing Customer Satisfaction Ratings Two different stores (Store X and Store Y) collected customer satisfaction ratings. We want to see if the satisfaction ratings are significantly different.

Sample data: Customer satisfaction ratings (out of 10)

store_x_ratings = [8, 9, 7, 8, 6, 9, 8, 7, 8, 9]
store_y_ratings = [7, 8, 6, 6, 7, 7, 8, 6, 7, 7]

In [23]:
# given sample space two stores
store_x_ratings = [8, 9, 7, 8, 6, 9, 8, 7, 8, 9] 
store_y_ratings = [7, 8, 6, 6, 7, 7, 8, 6, 7, 7]
#to calculate two independent sample t-test
t_stat,p_val=stats.ttest_ind(store_x_ratings,store_y_ratings)
print("t_statistic :",round(t_stat,4))
print("p_value :",round(p_val,5))
#interpretation
if p_val < 0.05:
    print("reject the null hypo : the mean of the two groups is  significantly different from each other")
else:
    print("fail to reject the null hypo : the mean of the two groups is not significantly different from each other")


t_statistic : 2.5538
p_value : 0.01994
reject the null hypo : the mean of the two groups is  significantly different from each other


#### prob 3:Comparing Reaction Times Between Two Age Groups Two groups of participants (under 30 and over 50) were tested for their reaction times to a stimulus. Determine if there is a significant difference between the two age groups.

#Sample data: Reaction times in milliseconds

under_30_times = [320, 310, 305, 315, 312, 318, 319, 314, 308, 311]

over_50_times = [400, 395, 405, 410, 392, 398, 403, 402, 396, 399]

In [26]:
# given sample space two stores
under_30_times = [320, 310, 305, 315, 312, 318, 319, 314, 308, 311] 
over_50_times = [400, 395, 405, 410, 392, 398, 403, 402, 396, 399]
#to calculate two independent sample t-test
t_stat,p_val=stats.ttest_ind(under_30_times,over_50_times)
print("t_statistic :",round(t_stat,4))
print("p_value :",round(p_val,5))
#interpretation
if p_val < 0.05:
    print("reject the null hypo : the mean of the two groups is  significantly different from each other")
else:
    print("fail to reject the null hypo : the mean of the two groups is not significantly different from each other")

t_statistic : -38.1623
p_value : 0.0
reject the null hypo : the mean of the two groups is  significantly different from each other


# paired t-test

#### prob 1:does the average performance of students change before and after a training program

In [96]:
#given sample space of two paired data
before_training=np.array([65,70,68,74,72,71,69])
after_training=np.array([78,75,80,79,77,76,82])
# to calculate paired t test
t_stat,p_val=stats.ttest_rel(before_training,after_training)
print("t_statistic :",round(t_stat,4))
print("p_value :",round(p_val,5))
#interpretation
if p_val < 0.05:
    print("reject the null hypo : there is  significantly difference from each other")
else:
    print("fail to reject the null hypo : there is no significant difference from each other")

t_statistic : -5.3318
p_value : 0.00178
reject the null hypo : there is  significantly difference from each other


#### prb 2:Weight Loss Program Effectiveness

Question: A fitness center wants to test the effectiveness of their new weight loss program. They recorded the weights of 10 participants before and after completing the program. Is there a significant weight change?

Sample data (weights before and after)

before_weights = [85, 90, 78, 88, 95, 102, 85, 91, 89, 100]

after_weights = [80, 85, 74, 84, 90, 98, 81, 87, 86, 96]

In [100]:
#given sample space for two paired data
before_wt=np.array([85,90,78,88,95,102,85,91,89,100])
after_wt=np.array([80,85,74,84,90,98,81,87,86,96])
#to calculate paired t-test
t_stat,p_val=stats.ttest_rel(before_wt,after_wt)
print("t_statistic :",round(t_stat,4))
print("p_value :",p_val)
#interpretation
if p_value < 0.05:
    print("reject the null hypo : there is  significantly difference from each other")
else:
    print("fail to reject the null hypo : there is no significant difference from each other")

t_statistic : 21.0
p_value : 5.901994059532877e-09
reject the null hypo : there is  significantly difference from each other


#### prb 3:Testing a New Diet Plan
Question: A researcher is testing the effectiveness of a new diet plan by comparing the cholesterol levels of 20 participants before and after following the diet for 3 months. Does the diet significantly affect cholesterol levels?

Sample data (cholesterol levels before and after)

before_cholesterol = [210, 220, 195, 240, 230, 205, 215, 225, 210, 235, 245, 200, 220, 215, 190, 225, 230, 195, 215, 205]

after_cholesterol = [190, 200, 185, 220, 210, 190, 200, 210, 195, 220, 230, 185, 200, 205, 175, 215, 220, 180, 200, 190]

In [107]:
#sample space of paired data
before_c=np.array([210,220,195,240,230,205,215,225,210,235,245,200,220,215,190,225,230,195,215,205])
after_c=np.array([190,200,185,220,210,190,200,210,195,220,230,185,200,205,175,215,220,180,200,190])
# to calculate the paired t - test
t_stat,p_val=stats.ttest_rel(before_c,after_c)
print("t_statistic :",round(t_stat,4))
print("p_value :",p_val)
#interpretation
if p_value < 0.05:
    print("reject the null hypo : there is  significantly difference from each other")
else:
    print("fail to reject the null hypo : there is no significant difference from each other")

t_statistic : 19.8738
p_value : 3.573174201316975e-14
reject the null hypo : there is  significantly difference from each other


#### prb 4:Effect of a Memory Enhancement Course
Question: A psychologist wants to assess if a memory enhancement course improves participants' scores on a memory test. She measures their scores before and after the course. Is there a significant improvement?

Sample data (memory test scores)

before_scores = [12, 15, 14, 11, 10, 13, 16, 14, 15, 12]

after_scores = [15, 18, 17, 14, 13, 15, 19, 17, 18, 15]

In [110]:
#sample space of paired data
before_me=np.array([12,15,14,11,10,13,16,14,15,12])
after_me=np.array([15,18,17,14,13,15,19,17,18,15])
#to calculate paired t-test
t_stat,p_val=stats.ttest_rel(before_me,after_me)
print("t_statistic :",round(t_stat,4))
print("p_value :",p_val)
#interpretation
if p_value < 0.05:
    print("reject the null hypo : there is  significantly difference from each other")
else:
    print("fail to reject the null hypo : there is no significant difference from each other")

t_statistic : -29.0
p_value : 3.3602799026670967e-10
reject the null hypo : there is  significantly difference from each other


#### prb 5:Evaluating Productivity Before and After a Workshop
Question: A manager wants to check if a recent productivity workshop has had an effect on employees' output. The number of units produced by employees is recorded before and after the workshop. Is there a significant difference?

Sample data (productivity in units)

before_output = [120, 125, 130, 110, 115, 140, 125, 130, 135, 120]

after_output = [130, 135, 140, 120, 125, 145, 130, 135, 140, 130]

In [115]:
#sample space of paired data
before_pw=np.array([120,125,130,110,115,140,125,130,135,120])
after_pw=np.array([130,135,140,120,125,145,130,135,140,130])
#to calculate paired t-test
t_stat,p_val=stats.ttest_rel(before_pw,after_pw)
print("t_statistic :",round(t_stat,4))
print("p_value :",p_val)
#interpretation
if p_value < 0.05:
    print("reject the null hypo : there is  significantly difference from each other")
else:
    print("fail to reject the null hypo : there is no significant difference from each other")

t_statistic : -9.798
p_value : 4.239738886259835e-06
reject the null hypo : there is  significantly difference from each other



# one sample t-test for proportions

#### prob 1: doees the average proportion of defect rates in a factory batch differ significantly from the benchmak of 5%

In [8]:
# sample data defect rates as percentage
defect_rate = np.array([4.5,5.2,5.1,4.8,5.5,4.9,5.0,4.7])
#convert into proportions
proportion=defect_rate/100
#to perform one - sample t test
t_stat,p_val=stats.ttest_1samp(proportion,0.05)

In [10]:
print("t_statistics:",t_stat)
print("p_value:",p_val)


t_statistics: -0.34062016887952185
p_value: 0.7433835895475498


In [14]:
#interpretation
if p_val < 0.05:
    print("reject the null hypo : the defect rate is significantly different from 5%")
else:
    print("fail to reject the null hypo : the defect rate is not significantly different from 5%")

fail to reject the null hypo : the defect rate is not significantly different from 5%


# z-test

In [5]:
!pip install statsmodels

In [5]:
import statsmodels.api as sm

#### 1. Testing Voter Support for a Candidate


Question: A political analyst claims that 60% of voters support a specific candidate. A random sample of 200 voters shows that 120 of them support the candidate. Is there enough evidence to refute the claim?

In [10]:
observed_proportion = 120/200

In [12]:
# given data
h0=0.06
n=200

In [14]:
#to perform z test 
ztest,p_val=sm.stats.proportions_ztest(120,n,h0)
print(f"t_statistics:{ztest},p_value{p_val}")

t_statistics:15.588457268119896,p_value8.72143960487895e-55


In [16]:
#interpration
if p_val <0.05:
    print("reject the null hypothesis")
else:
    print("fail to reject the null hypothesis")

reject the null hypothesis


#### prb 2: Employee Satisfaction Rate


Question: A company claims that 80% of its employees are satisfied with their jobs. A survey of 150 employees shows that 110 are satisfied. Is this claim accurate?

In [19]:
#given datas
ob_proportions=110/150
h0=0.80
n=150

In [21]:
#to perform z test
ztest,p_val=sm.stats.proportions_ztest(110,n,h0)
print(f"t_statistics:{ztest},p_value{p_val}")
#interpretation
if p_val <0.05:
    print("reject the null hypothesis")
else:
    print("fail to reject the null hypothesis")

t_statistics:-1.8463723646899934,p_value0.06483815699206606
fail to reject the null hypothesis


#### prb 3: Quality Check for a Product Line

Question: A manufacturer claims that 95% of products pass quality checks. In a random sample of 300 products, 280 passed the checks. Is the claim valid?


In [25]:
#given data
ob_proportions=280/300
h0=0.95
n=300

In [29]:
#to perform ztest
ztest,p_val=sm.stats.proportions_ztest(280,n,h0)
print(f"t_statistics:{ztest},p_value:{p_val}")
#interpretation
if p_val <0.05:
    print("reject the null hypothesis")
else:
    print("fail to reject the null hypothesis")

t_statistics:-1.1572751247156854,p_value:round(0.24715997321190875, 3)
fail to reject the null hypothesis


#### prb 4: Website Conversion Rate


Question: A website claims its conversion rate is 5%. Out of 500 website visits, 40 resulted in conversions. Does the data support this claim

In [32]:
#given datas
ob_proportions=40/500
h0=0.05
n=300
#to perform ztest
ztest,p_val=sm.stats.proportions_ztest(40,n,h0)
print(f"t_statistics:{ztest},p_value:{p_val}")
#interpretation
if p_val <0.05:
    print("reject the null hypothesis")
else:
    print("fail to reject the null hypothesis")

t_statistics:4.246038878042234,p_value:2.1758289255506334e-05
reject the null hypothesis


#### prb 5: Proportion of Customers Preferring a New Product

Question: A company claims that 70% of customers prefer their new product. A survey of 400 customers finds that 250 prefer the product. Test the claim.

In [37]:
#given data
ob_proportions=250/400
h0=0.70
n=400
#to perform ztest
ztest,p_val=sm.stats.proportions_ztest(250,n,h0)
print(f"t_statistics:{round(ztest,3)},p_value:{round(p_val,3)}")
#interpretation
if p_val <0.05:
    print("reject the null hypothesis")
else:
    print("fail to reject the null hypothesis")

t_statistics:-3.098,p_value:0.002
reject the null hypothesis


# welch's t-test
### unequal variance
are the average scores of two groups with unequal variance significatly different

In [47]:
# given data on scores of two groups
grp1_score=np.array([82,76,79,85,88,83])
grp2_score=np.array([72,65,70,60,75,68])
#to perform welch's t-test 
t_stat,p_val=stats.ttest_ind(grp1_score,grp2_score,equal_var=False)
print("t_statistic :",round(t_stat,4))
print("p_value :",round(p_val,4))
#interpretation
if p_val < 0.05:
    print("reject the null hypo : there is  significantly difference from each other")
else:
    print("fail to reject the null hypo : there is no significant difference from each other")

t_statistic : 4.9726
p_value : 0.0006
reject the null hypo : there is  significantly difference from each other


# point estimation

In [81]:
import random

#### consider the data of grade points for 20 students in a data science course. select grades of 10 students randomly from the data and find the point estimation for the population mean

In [58]:
# in sample space on grades of 20 students draw sample for 10 students without replacement
grade=[59.1, 65.0, 75.8, 79.2, 95.8, 99.8, 89.1, 05.2, 41.9, 55.2, 94.8, 84.1, 83.3, 74.0, 75.5, 76.2, 79.1, 80.1, 92.1, 74.2, 59.2, 64.0, 75, 78.2, 95.6, 97.8, 89.5, 64.2, 41.6, 57.2, 85, 91.4, 81.8, 74.6, 90]
#set a random seed to get the same sample each time you run the code
random.seed(1)
#pass the data to the parameter "population"
#pass the required sample size to the parameter ,'k'
samp_grades=random.sample(population=grade,k=20)
# calculating the sample mean
samp_mean=np.mean(samp_grades)
print("the point estimate for the population mean is ",samp_mean)

the point estimate for the population mean is  72.36


#### prb 2:a financial firm has created 50 portfolios for them , sample of 13 portfolios was selected , out of which 8 were found to be underperforming portfolios?

In [63]:
#population
N = 50
#sample
n = 13
#no. of underperforming portfolios
x = 8
#sample proportions
p_samp = x/n
#to estimate the no. of underperforming portfolios
num_port = p_samp*N
print("the no. of underperforming portfolios :",round(num_port))

the no. of underperforming portfolios : 31


## sampling error

1. Consider the data for the number of ice-creams sold per day. An ice-cream vendor collected this data for 90 days and then a sample is drawn (without replacement) containing ice-creams sold for 25 days.

data = (21, 93, 62, 76, 73, 20, 56, 95, 41, 36, 38, 13, 80, 88, 34, 18, 40, 11, 25, 29, 61, 23, 82, 10, 92, 69, 60, 87, 14, 91, 94, 49, 57, 83, 96, 55, 79, 52, 59, 39, 58, 17, 19, 98, 15, 54, 48, 46, 72, 45, 65, 28, 37, 30, 68, 75, 16, 33, 31, 99, 22, 51, 27, 67, 85, 47, 44, 77, 64, 97, 84, 42, 90, 70, 74, 89, 32, 26, 24, 12, 81, 53, 50, 35, 71, 63, 43, 86, 78, 66]

sample = (10, 22, 47, 66, 11, 57, 77, 98, 31, 63, 74, 84, 50, 96, 88, 92, 70, 54, 65, 44, 16, 72, 20, 90, 43]

In [69]:
data = [21, 93, 62, 76, 73, 20, 56, 95, 41, 36, 38, 13, 80, 88, 34, 18, 40, 11, 25, 29, 61, 23, 82, 10, 92, 69, 60, 87, 14, 91, 94, 49, 57, 83, 96, 55, 79, 52, 59, 39, 58, 17, 19, 98, 15, 54, 48, 46, 72, 45, 65, 28, 37, 30, 68, 75, 16, 33, 31, 99, 22, 51, 27, 67, 85, 47, 44, 77, 64, 97, 84, 42, 90, 70, 74, 89, 32, 26, 24, 12, 81, 53, 50, 35, 71, 63, 43, 86, 78, 66]
sample=[10,22,47,66,11,57,77,98,31,63,74,84,50,96,88,92,70,54,65,44,16,72,20,90,43]
#given sample
samp_mean= np.mean(sample)
#calculate the point estimate for the population mean
pop_mean=np.mean(data)
# population mean
print("sampleing error for mean :",np.abs(samp_mean-pop_mean))
# cal the sampling error for the mean

sampleing error for mean : 3.1000000000000014


## large sample size

#### prob 1)a random sample of weight for 35 diabetic patients is drawn from the population with a std deviation of 8kg.
find the 90% confidence interval for a population mean

In [79]:
#Given sample

weight=[59.1, 65.0, 75.8, 79.2, 95.8, 99.8, 89.1, 65.3, 41.9, 55.2, 94.8, 84.1, 83.2, 74.0, 75.5, 76.2, 79.1, 80.1, 92.1, 74.2, 59.2, 64.8, 75, 78.2, 95.6, 97.8, 89.5, 64.2, 41.8, 57.2, 85, 91.4, 81.8, 74.6, 90]

#number of observations in sample

n=len(weight)

#given population standard deviation

std_pop = 8

#calculate the 90% confidence intervol

#pass the sample mean to the parameter, 'Loc'

#pass the scaling factor (std_pop/n^(1/2)) to the parameter, 'scale'

interval= stats.norm.interval(0.90, loc= np.mean(weight), scale = std_pop/np.sqrt(n))

#print the 90% confidence interval

#use 'round()' to round-off the value to 2 digits
print("the 90% confidence interval of population mean in ",np.round(interval,2))

the 90% confidence interval of population mean in  [74.51 78.96]


#### prob 2)There are 150 apples on a tree. You randomly choose 40 apples and found that the average weight of apples is 182 grams with a standard deviation of 30 grams. Find the 95% confidence interval for the population mean.

In [91]:
#number of observations in the sample

n=40

#sample mean

sample_avg= 182

#sample standard deviation

sample_std=30

#calculate the 95% confidence interval

#pass the sample mean to the parameter, 'Loc'

#pass the scaling factor (sample std/n^(1/2)) to the parameter, 'scole'

#as the population standard deviation is unknown, use the sample standard deviation

interval=stats.norm.interval(0.95, loc=sample_avg, scale=sample_std/np.sqrt(n))

print("95% confidence Interval for population mean is",np.round(interval,2))

95% confidence Interval for population mean is [172.7 191.3]


#### prob3) Average Height of Students

Problem: A sample of 50 students has an average height of 170 cm with a standard deviation of 6 cm. Calculate the 95% confidence interval for the population mean.

In [89]:
#number of observations in the sample

n=50

#sample mean

sample_avg= 170

#sample standard deviation

sample_std=6

#calculate the 95% confidence interval

#pass the sample mean to the parameter, 'Loc'

#pass the scaling factor (sample std/n^(1/2)) to the parameter, 'scole'

#as the population standard deviation is unknown, use the sample standard deviation

interval=stats.norm.interval(0.95, loc=sample_avg, scale=sample_std/np.sqrt(n))

print("95% confidence Interval for population mean is",np.round(interval,2))

95% confidence Interval for population mean is [168.34 171.66]


#### prob4) Average Monthly Sales of a Product


Problem: A company samples 40 monthly sales records and finds an average of 5,000  with  a standard deviation of  1,200. Find the 95% confidence interval for the population mean monthly sales.

In [96]:
# no. of sample observations
n=40
# mean of sample
s_mean=5000
#std-deviation
s_std=1200
# to calculate the mean of the population 
interval = stats.norm.interval(0.95,loc=s_mean,scale=s_std/np.sqrt(n))
print("95% confidence interval for the population mean is ",np.round(interval,2))

95% confidence interval for the population mean is  [4628.12 5371.88]


#### prob5)Mean Temperature of a City


Problem: Meteorologists record the average daily temperature for 60 days in a city as 24°C with a standard deviation of 3°C. Calculate the 95% confidence interval for the population mean temperature.

In [99]:
# no. of sample observations
n=60
# mean of sample
s_mean=24
#std-deviation
s_std=3
# to calculate the mean of the population 
interval = stats.norm.interval(0.95,loc=s_mean,scale=s_std/np.sqrt(n))
print("95% confidence interval for the population mean is ",np.round(interval,2))

95% confidence interval for the population mean is  [23.24 24.76]


#### prob6) Average Time Spent on an App


Problem: A survey shows that 100 users spend an average of 45 minutes per day on an app with a standard deviation of 8 minutes. Determine the 95% confidence interval for the population mean time spent.

In [102]:
# no. of sample observations
n=100
# mean of sample
s_mean=45
#std-deviation
s_std=8
# to calculate the mean of the population 
interval = stats.norm.interval(0.95,loc=s_mean,scale=s_std/np.sqrt(n))
print("95% confidence interval for the population mean is ",np.round(interval,2))

95% confidence interval for the population mean is  [43.43 46.57]


#### prob 7) Mean Weight of Packages


Problem: A shipping company weighs 30 randomly selected packages and finds an average weight of 15.2 kg with a standard deviation of 2.1 kg. Compute the 95% confidence interval for the mean weight of all packages.

In [105]:
# no. of sample observations
n=30
# mean of sample
s_mean=15.2
#std-deviation
s_std=2.1
# to calculate the mean of the population 
interval = stats.norm.interval(0.95,loc=s_mean,scale=s_std/np.sqrt(n))
print("95% confidence interval for the population mean is ",np.round(interval,2))

95% confidence interval for the population mean is  [14.45 15.95]


# one sample t-test

#### prob1. The company's annual quality report of machines states that a lathe machine works efficiently at most for 8 months on average after the servicing. The production manager claims that after the special tuxan servicing, the machine works efficiently for more than 8 months. To test the claim of production manager consider the hypothesis:

In [14]:
# machine efficiency
month=np.array([2,4,3,6,5,10,7,8])
# sample size < 30 , hence T-test
# hence the critical value is not specified , we consider the default value of 0.05 . so , the confidence interval is 95%
#ho<=specific value : h1>specific value , from this we know it is one tailed test - right tailed test
t_stat,p_val=stats.ttest_1samp(month,8)
print("t_statistics :",round(t_stat,3))
print("p_value:",round(p_val,3))
# interpretation 
if p_val < 0.05:
    print("reject the null hypo : the machine efficiency is significantly different from 0.05")
else:
    print("fail to reject the null hypo : the machine efficiency is not significantly different from 0.05")


t_statistics : -2.517
p_value: 0.04
reject the null hypo : the machine efficiency is significantly different from 0.05


#### prob2. A railway authority claims that all the trains on the Chicago-Seattle route run with a speed of at least 56 mph on average. A customer forum declares that there are various records from passengers claiming that the speed of the train is less than what railway has claimed. In this scenario, a statistical test can be performed to test the claim of customer forum considering the hypothesis:

Null Hypothesis: Speed >= 56 mph

Alternative Hypothesis: Speed < 56 mph


In [17]:
# train speed 
speed=np.array([54,65,70,59,90,80,70,66,76])
# sample size < 30 , hence T-test
# hence the critical value is not specified , we consider the default value of 0.05 . so , the confidence interval is 95%
#ho>=specific value : h1<specific value , from this we know it is one tailed test - left tailed test
t_stat,p_val=stats.ttest_1samp(speed,54)
print("t_statistics :",round(t_stat,3))
print("p_value:",round(p_val,3))
# interpretation 
if p_val < 0.05:
    print("reject the null hypo (h1): Chicago-Seattle route doesn't run with a speed of at least 56 mph on average")
else:
    print("fail to reject the null hypo (h0) : Chicago-Seattle route run with a speed of at least 56 mph on average")

t_statistics : 4.396
p_value: 0.002
reject the null hypo (h1): Chicago-Seattle route doesn't run with a speed of at least 56 mph on average


### another way

In [73]:
def one_tail_ttest(data,mean,alpha=0.05):
    from scipy import stats
    t_statistic,p_value = stats.ttest_1samp(data,mean)

    decision = "reject ho" if p_value<alpha else "fail to reject ho"
    return t_statistic,p_value,decision

#### 3. A housing authority claims that the average monthly rent for a 1-bedroom apartment in a city is $1,500. A sample of 25 tenants' rents is taken, and the following data is recorded (in dollars)


rents = [1520, 1490, 1480, 1510, 1530, 1470, 1505, 1485, 1495, 1500,
         1525, 1515, 1495, 1475, 1510, 1480, 1520, 1495, 1500, 1485,
         1535, 1490, 1515, 1480, 1505, 1515, 1490]

In [33]:
# monthly rent
data=np.array([1520, 1490, 1480, 1510, 1530, 1470, 1505, 1485, 1495, 1500, 1525, 1515, 1495, 1475, 1510, 1480, 1520, 1495, 1500, 1485, 1535, 1490, 1515, 1480, 1505, 1515, 1490])
mean=1500
# population mean = 1500
# sample size < 30 , hence T-test
# hence the critical value is not specified , we consider the default value of 0.05 . so , the confidence interval is 95%
def one_sample_ttest(data,mean,alpha=0.05):
    from scipy import stats
    t_statistic,p_value = stats.ttest_1samp(data,mean)

    decision = "reject the null hypo (h1):" if p_value<alpha else "fail to reject (ho)"
    return t_statistic,p_value,decision

In [35]:
one_sample_ttest(data,mean,alpha=0.05)

(0.16439898730538718, 0.8706884780268073, 'fail to reject (ho)')

#### 2. Customer Satisfaction in a Retail Store
A retail store claims that 90% of their customers are satisfied with their shopping experience. A survey is conducted with 120 customers, and the satisfaction scores (out of 100) are as follows

satisfaction_scores = [92, 85, 88, 90, 91, 87, 86, 84, 93, 89,
                       95, 92, 91, 88, 87, 90, 92, 85, 89, 93,
                       94, 90, 88, 91, 93, 87, 89, 85, 88, 86,
                       92, 90, 89, 92, 87, 91, 88, 92, 90, 85,
                       93, 87, 91, 92, 89, 86, 94, 91, 88, 92,
                       90, 91, 93, 90, 86, 87, 89, 91, 88, 92,
                       90, 85, 94, 91, 92, 93, 88, 90, 91, 89]

# z test two tail 

### A car manufacturing company claims that t the mileage of their new car is 25 kmph with a standard deviation of 2.5 kmph. A random sample of 45 cars was drawn and recorded their mileage e as per the standard procedure. From the sample, the mean mileage was seen to be 24 kmph. Is this evidence to claim that the mean mileage is different from 25kmph? (assume the normality of the data) Use a = 0.01.

##### calculating z - value

In [9]:
#car manufacturing 
# cal the z-value 99% for confidence level 
# use 'stats.norm.isf()',to find the z-value corresponding
# to upper tail probability (q)
# pass the value of 'alpha/2' to the parameter 'q' here alpha=0.01
# use'round()' to round off the value to 2 digits

z_val=np.abs(round(stats.norm.isf(q=0.01/2),2))
print('critical value for two-tailed z-test:',z_val)

# if z is less than -2.58 or greater than 2.58 then we reject the null hypo

critical value for two-tailed z-test: 2.58


##### calculating z-score

In [11]:
# define a function to calculate the z-test statistic
# pass the population mean,population std,sample size and sample mean as the function input

def z_test(pop_mean,pop_std,n,samp_mean):
    z_score=(samp_mean-pop_mean)/(pop_std/np.sqrt(n)) 
    # calculate the z-test statistic

# return the z-test value
    return z_score

# given data 
n=45
pop_mean=25
pop_std=2.5
samp_mean=24

# calculate the statistic using the function 'z-test'

z_score=z_test(pop_mean,pop_std,n,samp_mean)
print('z-score:',z_score)


z-score: -2.6832815729997477


##### calculating p value

In [61]:
#calculate the corresponding p value for the test statistics
#use cdf() to calculate p(Z<=z_score)

p_value=stats.norm.cdf(z_score)

#for a two_tailed test multiply the p value by 2

req_p=p_value*2
print('p_value:',req_p)

p_value: 1.109428736177261


##### calculating confidence interval

In [63]:
#calculate the 99%confidence interval for the population mean
#pass the sample mean to the parameter,'loc'
#pass the scaling factor (pop_std/n^(1/2)) to the parameter ,'scale'

print('confidence interval:',stats.norm.interval(.99,loc=samp_mean,scale=pop_std/np.sqrt(n)))

confidence interval: (2.911388891226528, 3.0986111087734716)


`here the z score is less than -2.58, p_value is less than .01, also the confidence interval do not contain the value in the null hypothesis(ie 25),thus we reject the null hypothesis and conclude that there is enough evidence to claim that the mean mileage is differnt from 25 kmph`

#### The average calories in a slice bread of the brand 'Alphas' are 82 with a standard deviation of 15. An experiment is conducted to test the claim of the dietitians that the calories in a slice of bread are not as per the manufacturer's specification. A sample of 40 slices of bread is taken and the mean calories recorded are 95. Test the claim of dietitians with a value (significance level) as 0.05. (assume the normality of the data).

In [22]:
# to cal the normality 
# shapiro test
# given alpha is 0.05
sample_d=[102.45, 92.93, 104.72, 117.85, 91.49, 91.49, 118.69, 106.51, 87.96, 103.14, 
 88.05, 88.01, 98.63, 66.30, 69.13, 86.57, 79.81, 99.71, 81.38, 73.82, 
 116.98, 91.61, 96.01, 73.63, 86.83, 96.66, 77.74, 100.64, 85.99, 90.62, 
 85.97, 122.78, 94.80, 79.13, 107.34, 76.69, 98.13, 65.60, 75.08, 97.95]
n=len(sample_d)
print(n)
alpha=0.05
stat,p_val=stats.shapiro(sample_d)
print(f"statistics:{round(stat,3)}")
print(f"p_value:{round(p_val,3)}")
print('\n')
if p_val < 0.05:
    print("reject the null hypo (h1): the given value is not normally distribution")
else:
    print("fail to reject the null hypo (h0) :the given value is normally distribution")

40
statistics:0.979
p_value:0.661


fail to reject the null hypo (h0) :the given value is normally distribution


In [14]:
# have to cal confidence level of 95%
# have to find z-level correspondence of the upper tail(q)
z_val=np.abs(round(stats.norm.isf(q=0.05/2),2))
# q is taken as 0.05/2 as the given alpha value=0.05
print('critical value for two-tailed z-test:',z_val)

critical value for two-tailed z-test: 1.96


`if z is less than -1.96 or greater than 1.96 ,then we reject the null hypothesis`

In [30]:
# z-test statistics & z-score
# required values : no of sample,population mean , poppulation std deviation,sample mean
# call the function z-test()
def z_test(pop_mean,pop_std,n,samp_mean):
    z_score=(samp_mean-pop_mean)/(pop_std/np.sqrt(n)) 
    # calculate the z-test statistic

# return the z-test value
    return z_score
n=40
pop_mean=82
pop_std=15
samp_mean=95

z_score=z_test(pop_mean,pop_std,n,samp_mean)
print('z-score:',z_score)


z-score: 5.4812812776251905


In [39]:
# to cal the corresponding p_val for the test stat
# use cdf to cal p(z<=zscore)
p_val=stats.norm.cdf(-z_score)#try norm.sf(z_score) also gives the same val
# for a 2 tail test multily the p_value by 2
req_p=p_val*2
print('p_val:',req_p)

p_val: 4.222565249683579e-08


In [44]:
4.222565249683579e-08<0.05

True

In [34]:
#cal the 95% confidence interval for pop mean 
#pass the sample mean to te parameter , 'loc'
# pass the scaling factor (pop_std/n^(1/2)) to the parameter,'scale'
print("confidence interval:",stats.norm.interval(0.95,loc=samp_mean,scale=pop_std/np.sqrt(n)))

confidence interval: (90.35153725771579, 99.64846274228421)


`here the z score is greater than 1.96 , p-value is  less the 0.05 and the confidence interval do not contain the value in the null hypo(i.e 82).thus we reject the ho . there is enough evidence to conclude that the avg calories in a slice bread are not 82.`

#### 3. A typhoid vaccine in the market inscribes 3 mg mg of f ascorbic acid in the vaccine. A research team claims that the vaccines contain less than 3 3 mg of acid. We collected the data of 40 vaccines by using random sampling from a population and recorded the amount of ascorbic acid. Test the claim of the research team using the sample data a value (significance level) to 0.05.

acid_amt=[2.57, 3.06, 3.28, 3.24, 2.79, 3.48, 3.36, 3.07, 2.46, 3.03, 3.05, 2.94, 3.46, 3.19, 3.09, 2.81, 3.13, 2.88, 2.76, 2.75, 3.17, 2.89, 2.54, 3.18, 3.08, 2.60, 3.06, 3.13, 3.11, 3.08, 2.93, 2.90, 3.06, 2.97, 3.24, 2.86, 2.87, 3.18, 3, 2.95]

Let us check the normality of the data.

In [82]:
from statsmodels.stats import weightstats as stests
# it is ho >= 3 :h1 < 3 
# hence it is one tail - left
# to cal the normality 
# shapiro test
# given alpha is 0.05
sample_d=[2.57, 3.06, 3.28, 3.24, 2.79, 3.48, 3.36, 3.07, 2.46, 3.03, 3.05, 2.94, 3.46, 3.19, 3.09, 2.81, 3.13, 2.88, 2.76, 2.75, 3.17, 2.89, 2.54, 3.18, 3.08, 2.60, 3.06, 3.13, 3.11, 3.08, 2.93, 2.90, 3.06, 2.97, 3.24, 2.86, 2.87, 3.18, 3, 2.95]
n=len(sample_d)
print(n)
alpha=0.05
stat,p_val=stats.shapiro(sample_d)
print(f"statistics:{round(stat,3)}")
print(f"p_value:{round(p_val,3)}")
print('\n')
if p_val < 0.05:
    print("reject the null hypo (h1): the given value is not normally distribution")
else:
    print("fail to reject the null hypo (h0) :the given value is normally distribution")

40
statistics:0.977
p_value:0.588


fail to reject the null hypo (h0) :the given value is normally distribution


In [73]:
# have to cal confidence level of 95%
# have to find z-level correspondence of the upper tail(q)
z_val=np.abs(round(stats.norm.isf(q=0.05),2))# here 0.05/2 no need since one tail
# q is taken as 0.05/2 as the given alpha value=0.05
print('critical value for two-tailed z-test:',z_val)

critical value for two-tailed z-test: 1.64


`if z is less than -1.64 or greater than 1.64 ,then we reject the null hypothesised
`

In [53]:
#ztest() returns the test statistics and corresponding p-val
#pass the sample data

z-score: 0.13758139056882976


In [56]:
# to cal the corresponding p_val for the test stat
# use cdf to cal p(z<=zscore)
#p_val=stats.norm.cdf(-z_score)#try norm.sf(z_score) also gives the same val
# for a 2 tail test multily the p_value by 2
#req_p=p_val*2
#print('p_val:',req_p)
z_score,p_val=stats.ztest(x1=)

p_val: 0.890571263822739


In [58]:
#cal the 95% confidence interval for pop mean 
#pass the sample mean to te parameter , 'loc'
# pass the scaling factor (pop_std/n^(1/2)) to the parameter,'scale'
print("confidence interval:",stats.norm.interval(0.95,loc=samp_mean,scale=pop_std/np.sqrt(n)))

confidence interval: (2.9337707452135584, 3.0762292547864414)


`here the z score is greater than 1.96 , p-value is greater the 0.05 and the confidence interval do contain the value in the null hypo(i.e 3).thus we fail to reject the ho . there is enough evidence to conclude that the A typhoid vaccine in the market inscribes 3 mg mg of f ascorbic acid in the vaccine.`

#### 4) A sample of 900 PVC pipes is found to have an average thickness of 12.5 mm. Can we assume that the sample is coming from a normal population with mean 13 mm against that it is less than 13 mm. The population standard deviation is 1 mm. Test the hypothesis at 5% level of significance.

In [7]:
# to cal the normality 
# shapiro test - can't be done hence the sample space is 900
# given alpha is 0.05
# have to cal confidence level of 95%
# have to find z-level correspondence of the upper tail(q)
z_val=np.abs(round(stats.norm.isf(q=0.05),2))# here 0.05/2 no need since one tail
# q is taken as 0.05/2 as the given alpha value=0.05
print('critical value for two-tailed z-test:',z_val)


critical value for two-tailed z-test: 1.64


In [18]:
## shapiro wilk test
import pandas as pd
from scipy import stats


In [43]:
data=[45,88,12,44,90,33,56,10]
alpha=0.05
stat,p_val=stats.shapiro(data)
print(f"statistics:{round(stat,3)}")
print(f"p_value:{round(p_val,3)}")
print('\n')
if p_val < 0.05:
    print("reject the null hypo (h1): the given value is not normally distribution")
else:
    print("fail to reject the null hypo (h0) :the given value is normally distribution")


statistics:0.913
p_value:0.377


fail to reject the null hypo (h0) :the given value is normally distribution


#### 5. An e-commerce company claims that the mean delivery time of food items on its website in NYC is 60 minutes with a standard deviation of 30 minutes. A random sample of 45 customers ordered from the website, and the average time for delivery was found to be 75 minutes. Is this enough evidence to claim that the average time to get items delivered is more than 60 minutes. (assume the normality of the data). Test the with alpha = 0.05

The null and alternative hypothesis is:

H 0 dot alpha*mu <= 80

H_{1} / mu > 60

Here alpha = 0.05 for a one-tailed test calculate the critical z-value.

# interval estimation

In [8]:
# monthly expenditure
mean=300
std=50
sample=144
alpha=0.1
scale=std/np.sqrt(sample)
#number of observations in sample

#given population standard deviatio

#calculate the 90% confidence intervol

#pass the sample mean to the parameter, 'Loc'

#pass the scaling factor (std_pop/n^(1/2)) to the parameter, 'scale'

interval= stats.norm.interval(0.90, loc= mean, scale = std/np.sqrt(sample))

#print the 90% confidence interval

#use 'round()' to round-off the value to 2 digits
print("the 90% confidence interval of population mean in ",np.round(interval,2))

the 90% confidence interval of population mean in  [293.15 306.85]


#### A factory claims its machines produce an average of 200 units per day. After upgrading a machine, a manager tests whether the machine now produces more than 200 units on average. A sample of 30 days’ production is collected

production_data = [210, 215, 202, 205, 230, 220, 210, 208, 215, 214,
                   218, 220, 225, 222, 210, 219, 218, 230, 240, 235,
                   250, 245, 260, 255, 212, 218, 222, 210, 208, 250]

In [17]:
# production
production_data = [210, 215, 202, 205, 230, 220, 210, 208, 215, 214, 218, 220, 225, 222, 210, 219, 218, 230, 240, 235, 250, 245, 260, 255, 212, 218, 222, 210, 208, 250]
# given sample data for 30 days
n=len(production_data)
print("no. of sample :",n)
# if n is >= 30 , then we go with z-test
production_mean=200
# it is given the average per day is 200 units .so, the mean is 200
alpha=0.05
# nither confidence interval is given nor the p-value . therefore the default value is considered
# as it is not mentioned whether is it normality or not. we are going with shapiro
stat,p_val=stats.shapiro(production_data)
print(f"statistics:{round(stat,3)},p_value:{round(p_val,3)}")
# interpretation
if p_val < 0.05:
    print("reject the null hypo (h1): machine now produces more than 200 units on average")
else:
    print("fail to reject the null hypo (h0) :machine doesn't produces more than 200 units on average")



no. of sample : 30
statistics:0.893,p_value:0.006
reject the null hypo (h1): machine now produces more than 200 units on average


# another method

Two sales teams claim to have similar sales performances. You want to test whether Team A has significantly greater average weekly sales than Team B.


team_a_sales = [550, 600, 580, 590, 605, 620, 630, 615, 610, 620]
team_b_sales = [500, 520, 510, 530, 540, 525, 535, 515, 505, 520]

In [37]:
# one - right tail test
# independent two test - t test
team_a_sales = [550, 600, 580, 590, 605, 620, 630, 615, 610, 620]
team_b_sales = [500, 520, 510, 530, 540, 525, 535, 515, 505, 520]
# sample data of two teams
t_stat,p_val=stats.ttest_ind(team_a_sales,team_b_sales)
# to perform indeoendent 2 test
print(f"t_statistics:{round(t_stat,3)},p_value:{p_val}")
# interpretation
if p_val < 0.05:
    print("reject the null hypo : there is  significantly difference from each other")
else:
    print("fail to reject the null hypo : there is no significant difference from each other")

t_statistics:9.642,p_value:1.561707880255758e-08
reject the null hypo : there is  significantly difference from each other


Suppose you have a sample of test scores and want to test if the average score is equal to 75. Before performing the Z-test, you need to check if the data follows a normal distribution.

Steps to be included :
1. Check normality using a Shapiro-Wilk test (a common test for normality).

2. Perform the Z-test if the normality assumption holds.


Null Hypothesis (H0): The population mean is 75.


Alternative Hypothesis (H1): The population mean is not 75 (two-tailed).

In [50]:
# sample test score
test_s=np.array([72, 73, 74, 75, 76, 77, 78, 79, 80, 81,70, 71, 72, 73, 74, 75, 76, 77, 78, 79,69, 70, 71, 72, 73, 74, 75, 76, 77, 78])
print(len(test_s))
# if n is >= 30 , then we go with z-test
t_mean=np.mean(test_s)
t_std=np.std(test_s)
n=30
ho=75
# it is given the average per day is 200 units .so, the mean is 200
alpha=0.05
# nither confidence interval is given nor the p-value . therefore the default value is considered
# as it is not mentioned whether is it normality or not. we are going with shapiro
stat,p_val=stats.shapiro(test_s)
# z-test
scale= t_std/np.sqrt(n)
z_test=(t_mean-ho)/(scale)
print('z_statistics:',z_test)
# t_stat,p_val
print(f"statistics:{round(stat,3)},p_value:{round(p_val,3)}")

# interpretation
if p_val < 0.05:
    print("reject the null hypo (h1): Alternative Hypothesis (H1): The population mean is not 75 (two-tailed)")
else:
    print("fail to reject the null hypo (h0) :Null Hypothesis (H0): The population mean is 75.")



30
z_statistics: -0.2915233021459105
statistics:0.977,p_value:0.744
fail to reject the null hypo (h0) :Null Hypothesis (H0): The population mean is 75.


#### You want to compare the average test scores of two classes: Class A and Class B. You randomly sample 30 students from each class and get the following test scores:

Class A (30 students): [70, 75, 80, 85, 90, 72, 88, 84, 79, 77, 68, 76, 91, 85, 80, 73, 82, 78, 79, 86, 81, 85, 72, 74, 89, 90, 76, 83, 75, 80]


Class B (30 students): [68, 74, 79, 70, 82, 78, 74, 75, 80, 76, 68, 72, 83, 79, 85, 77, 74, 69, 80, 82, 71, 78, 73, 79, 81, 76, 84, 75, 70, 73]


Null Hypothesis (H0): The mean test scores of Class A and Class B are equal.


Alternative Hypothesis (H1): The mean test scores of Class A and Class B are different

In [64]:
# sample data of 2 classes
Class_A =np.array ([70, 75, 80, 85, 90, 72, 88, 84, 79, 77, 68, 76, 91, 85, 80, 73, 82, 78, 79, 86, 81, 85, 72, 74, 89, 90, 76, 83, 75, 80])
Class_B =np.array ([68, 74, 79, 70, 82, 78, 74, 75, 80, 76, 68, 72, 83, 79, 85, 77, 74, 69, 80, 82, 71, 78, 73, 79, 81, 76, 84, 75, 70, 73])# to get the no. of samples in the above data
print(f"count_a:{len(Class_A)},count_b:{len(Class_B)}")
# to perform independent 2 test
t_stat,p_val=stats.ttest_ind(Class_A,Class_B)
# t_stat,p_val
print(f"statistics:{round(t_stat,3)},p_value:{round(p_val,3)}")

# interpretation
if p_val < 0.05:
    print("reject the null hypo (h1):The mean test scores of Class A and Class B are different ")
else:
    print("fail to reject the null hypo (h0) :The mean test scores of Class A and Class B are equal")



count_a:30,count_b:30
statistics:2.704,p_value:0.009
reject the null hypo (h1):The mean test scores of Class A and Class B are different 


#### You want to compare the average weight loss after 3 months of using two different exercise regimens: Regimen A and Regimen B. You randomly select 50 participants from each regimen and measure their weight loss

Regimen A (50 participants): [3.5, 4.0, 3.2, 3.8, 4.3, 3.0, 3.7, 3.5, 4.1, 4.0, 3.9, 3.7, 4.2, 3.8, 3.9, 3.4, 3.3, 3.6, 3.8, 4.0, 3.6, 4.1, 3.7, 3.5, 4.2, 3.8, 3.7, 3.9, 4.0, 4.1, 3.5, 3.6, 3.7, 4.0, 4.1, 3.8, 3.9, 3.6, 4.0, 3.7, 3.5, 4.0, 3.8]


Regimen B (50 participants): [4.0, 4.5, 4.2, 4.3, 4.6, 4.4, 4.0, 4.1, 4.2, 4.4, 4.1, 4.3, 4.5, 4.2, 4.6, 4.0, 4.4, 4.2, 4.1, 4.3, 4.4, 4.6, 4.0, 4.2, 4.3, 4.1, 4.5, 4.4, 4.2, 4.1, 4.0, 4.3, 4.5, 4.1, 4.4, 4.2, 4.6, 4.3, 4.4, 4.0, 4.2, 4.1, 4.3, 4.5, 4.4, 4.2, 4.3]



Null Hypothesis (H0): The mean weight loss for Regimen A and Regimen B are equal.


Alternative Hypothesis (H1): The mean weight loss for Regimen A and Regimen B are different.

In [69]:
# given sample for 2 regimen
Regimen_A=np.array([3.5, 4.0, 3.2, 3.8, 4.3, 3.0, 3.7, 3.5, 4.1, 4.0, 3.9, 3.7, 4.2, 3.8, 3.9, 3.4, 3.3, 3.6, 3.8, 4.0, 3.6, 4.1, 3.7, 3.5, 4.2, 3.8, 3.7, 3.9, 4.0, 4.1, 3.5, 3.6, 3.7, 4.0, 4.1, 3.8, 3.9, 3.6, 4.0, 3.7, 3.5, 4.0, 3.8])

Regimen_B=np.array([4.0, 4.5, 4.2, 4.3, 4.6, 4.4, 4.0, 4.1, 4.2, 4.4, 4.1, 4.3, 4.5, 4.2, 4.6, 4.0, 4.4, 4.2, 4.1, 4.3, 4.4, 4.6, 4.0, 4.2, 4.3, 4.1, 4.5, 4.4, 4.2, 4.1, 4.0, 4.3, 4.5, 4.1, 4.4, 4.2, 4.6, 4.3, 4.4, 4.0, 4.2, 4.1, 4.3, 4.5, 4.4, 4.2, 4.3])
print(f"count_a:{len(Regimen_A)},count_b:{len(Regimen_B)}")
# to perform independent 2 test
t_stat,p_val=stats.ttest_ind(Regimen_A,Regimen_B)
# t_stat,p_val
print(f"statistics:{round(t_stat,3)},p_value:{p_val}")

# interpretation
if p_val < 0.05:
    print("reject the null hypo (h1):The mean weight loss of Regimen_A and Regimen_B are different ")
else:
    print("fail to reject the null hypo (h0) :The mean weight loss of Regimen_A and Regimen_B are equal")

count_a:43,count_b:47
statistics:-10.058,p_value:2.736122123225712e-16
reject the null hypo (h1):The mean weight loss of Regimen_A and Regimen_B are different 


## Sampling Error for Mean Age in a Population


You want to estimate the sampling error for the mean age of a population of 1000 individuals. You randomly sample 50 individuals, and their mean age is 35 years. The population mean age is 37 years, and the standard deviation of the population is 10 years.



Population Mean Age: 37 years
Sample Mean Age: 35 years
Sample Size (n): 50
Population Standard Deviation (σ): 10 years

In [72]:
n_data = 1000
sample_n=50
s_mean=35
p_mean=37
std=10

print("sampleing error for mean :",np.abs(s_mean-p_mean))
# cal the sampling error for the mean

sampleing error for mean : 2


## Sampling Error for Proportion in a Population


You want to estimate the sampling error for the proportion of people in a population who prefer a certain brand. A sample of 200 people is taken, and 120 people in the sample prefer the brand. The population proportion is 0.65.

Population Proportion (p): 0.65
Sample Proportion (p̂): 120/200 = 0.60
Sample Size (n): 200

In [79]:
#given sample
sample_n=200
s_proportion=0.60
p_proportion=0.65
# population mean
print("sampleing error for mean :",np.abs(s_proportion-p_proportion))
# cal the sampling error for the mean

sampleing error for mean : 0.050000000000000044


## Estimating Population Mean with a Small Sample


You are conducting a survey to estimate the average salary of employees at a small company. The sample consists of 15 employees, and their salaries (in thousands of dollars) are as follows


[45,52,48,47,53,49,50,46,51,52,50,48,49,46,45]


Use the sample mean as a point estimate for the population mean salary.

In [85]:
salary=[45,52,48,47,53,49,50,46,51,52,50,48,49,46,45]
# draw sample for salary without replacement
# set random seed to get the same sample each time you run the code
random.seed(1)

samp_salary=random.sample(population=salary,k=15)
samp_mean=np.mean(samp_salary)
print("the point estimate for the population mean is ",samp_mean)

the point estimate for the population mean is  48.733333333333334


## Estimating Population Proportion with a Small Sample


`A researcher conducts a poll of 25 people and finds that 15 out of the 25 people support a certain policy. Estimate the population proportion of support for the policy using the sample proportion.`

In [1]:
pop=25
sample=15
# to cal point estimation sample/population
ob_proportion=15/25
# population proportion is same as sample proportion 
pop_proportion=25/15
#to cal sampling error 
samp_error=pop_proportion-ob_proportion
# interpretation
print("The point estimate of the population proportion is :",ob_proportion)
print("The actual population proportion is :",pop_proportion)
print("The sampling error of the sample proportion is :",samp_error)


The point estimate of the population proportion is : 0.6
The actual population proportion is : 1.6666666666666667
The sampling error of the sample proportion is : 1.0666666666666669
